### DISCUSSION TITLE
ThunderClient - is there a way to traverse the xml responses included in the json response body and apply validations

### DISCUSSION THREAD

**Discussion by DeltaUser121** [thunderclient/thunder-client-support#1231] [Q&A]
ThunderClient - is there a way to traverse the xml responses included in the json response body and apply validations 

eg:
Json response as below

{
 "isDuplicate": null,
"message": "
<img width="578" alt="image" src="https://github.com/rangav/thunder-client-support/assets/139353595/1eb1cd0a-cfdf-44f7-94c8-cfc9419ae516">
"
}

If I want to traverse the message and want to get the value of node ContactId , how to do that?

<img width="578" alt="image" src="https://github.com/rangav/thunder-client-support/assets/139353595/3efe7430-3dfa-449c-8720-45da02cb225c">

**Comment by rangav** [comment]
Hi @DeltaUser121 you can use custom scripting and Cheerio library to parse xml.

Custom Scripting Docs
https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md

Let me know if you any further questions
